lightningd/peer_control.c

```diff
@@ -1437,13 +1437,14 @@ static bool check_funding_details(const struct bitcoin_tx *tx,
 				  struct amount_sat funding,
 				  u32 funding_outnum)
 {
-	struct amount_asset asset =
-	    bitcoin_tx_output_get_amount(tx, funding_outnum);
+	struct amount_asset asset;
 
-	if (!amount_asset_is_main(&asset))
+	if (funding_outnum >= tx->wtx->num_outputs)
```

Comment:
do you think that in this case can be necessary to add a log line at debug level? or this check will be pretty clear to debug if happens ?